esp/clients/ws_dfsclient/ws_dfsclient.cpp

```diff
@@ -556,16 +561,64 @@ IClientWsDfs *getDfsClient(const char *serviceUrl, IUserDescriptor *userDesc)
     return dfsClient.getClear();
 }
 
+static CriticalSection localSecretCrit;
+static void configureClientSSL(IEspClientRpcSettings &rpc, const char *secretName)
+{
+    /*
+     * This is a bit of a kludge, it gets the certificates from secrets, and writes them to local temp strorage.
+     * It does this so that it can pass the filename paths to rpc ssl / secure socket layer, which currently only
+     * accepts filenames, not binary blobs from memory.
+     */
+    StringBuffer clientCertFilename, clientPrivateKeyFilename, caCertFilename;
+
+    StringBuffer tempDirStr;
+    verifyex(getConfigurationDirectory(getGlobalConfigSP()->queryPropTree("Directories"), "temp", "ssl", "ssl", tempDirStr));
+    addPathSepChar(tempDirStr);
+    tempDirStr.append(secretName);
+    addPathSepChar(tempDirStr);
+
+    clientCertFilename.append(tempDirStr).append("tls.crt");
+    clientPrivateKeyFilename.append(tempDirStr).append("tls.key");
+    caCertFilename.append(tempDirStr).append("ca.crt");
+
+    CriticalBlock b(localSecretCrit);
+    if (!checkDirExists(tempDirStr.str()))
+    {
+        Owned<IFile> dir = createIFile(tempDirStr.str());
+        dir->createDirectory();
+        StringBuffer secretValue;
+
+        Owned<IFile> file = createIFile(clientCertFilename);
+        Owned<IFileIO> io = file->open(IFOcreate);
+        getSecretValue(secretValue, "storage", secretName, "tls.crt", true);
+        io->write(0, secretValue.length(), secretValue.str());
+
```

Comment:
minor:  Should calls to write() be followed by io->close() to ensure flushed, and avoid any issues with disk full throwing exceptions??

Reply:
agree, would be better. I'll add the io->close()'s